pkg/server/commands/list_objects.go

```diff
@@ -201,14 +201,20 @@ func (q *ListObjectsQuery) evaluate(
 		}
 
 		connectedObjectsResChan := make(chan *connectedobjects.ConnectedObjectsResult, 1)
-		var objectsFound = new(uint32)
 
 		connectedObjectsQuery := connectedobjects.NewConnectedObjectsQuery(q.datastore, typesys,
 			connectedobjects.WithResolveNodeLimit(q.resolveNodeLimit),
 			connectedobjects.WithResolveNodeBreadthLimit(q.resolveNodeBreadthLimit),
 			connectedobjects.WithMaxResults(maxResults),
 		)
 
+		// foundObjects map allows us to not return the same object twice
+		var foundObjectsMap sync.Map
+		var foundCount *uint32
+		if maxResults > 0 {
```

Comment:
If `maxResults` == 0 then that's supposed to indicate "no limit". Maybe it would be better to change this to a `*uint32` so that if it is nil then we treat it as no limit 🤷 ?

Reply:
> If maxResults == 0 then that's supposed to indicate "no limit". 

Correct

Reply:
In this case if maxResults==0 indicates no limit, then https://github.com/openfga/openfga/pull/921/files#r1300329789 doesn't make sense for me.

Reply:
I can't follow that link, but does it make more sense now?

Reply:
Yes, it makes more sense now.